flatten.go

```diff
@@ -0,0 +1,167 @@
+package authzed
+
+import (
+	"fmt"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+)
+
+// FlattenExpand reduces an ExpandResponse into the slice of Users present in
+// the expansion.
+//
+// Notably, this removes the context of which relations caused users to be
+// included in the expansion, but often you only need to know which users are
+// present.
+func FlattenExpand(resp *api.ExpandResponse) []*api.User { return flatten(resp.TreeNode) }
+
+func flatten(node *api.RelationTupleTreeNode) []*api.User {
+	switch typed := node.NodeType.(type) {
+	case *api.RelationTupleTreeNode_IntermediateNode:
+		switch typed.IntermediateNode.Operation {
+		case api.SetOperationUserset_UNION:
+			return flattenUnion(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_INTERSECTION:
+			return flattenIntersection(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_EXCLUSION:
```

Comment:
Add a warning if we encounter an unknown type?

Reply:
I've changed the signature to return an error. When we figure out the desirable way to handle future gRPC API changes, we can use that error.